homeassistant/components/webhook/__init__.py

```diff
@@ -117,6 +119,20 @@ async def async_handle_webhook(
             _LOGGER.warning("Received remote request for local webhook %s", webhook_id)
             return Response(status=HTTPStatus.OK)
 
+        # The presence of NON_LOCAL_HEADERS could indicate a CSRF request from a
```

Comment:
I think that this misses the use case in which a user hosts a local website to provide guests with a basic webhook powered UI.

Reply:
(cross-origin is required for that too)

Reply:
WDYT about checking the referer/origin URLs against the cors_allowed_origins configuration in the http component? If the URLs match, then consider it okay for a local_only webhook. That would allow a way to permit this use case without allowing all of the internet to access the webhook.

Reply:
Yeah I can't remember why CORS was allowed to True, but I agree that since it can opt-in origins via the `http` config, it's fine to disable it.

Reply:
Sounds good. Thank you. I've done that, and also added a referer header check for HEAD/POST [requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests). I also updated the description to consider this PR a breaking change, just in the off chance that someone was relying on this behavior.

Can you add the breaking change label to this PR so that it'll show-up in the release notes?